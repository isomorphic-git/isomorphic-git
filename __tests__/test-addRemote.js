/* eslint-env node, browser, jasmine */
const { makeFixture } = require('./__helpers__/FixtureFS.js')
// @ts-ignore
const snapshots = require('./__snapshots__/test-addRemote.js.snap')
const registerSnapshots = require('./__helpers__/jasmine-snapshots')
const { addRemote, listRemotes } = require('isomorphic-git')

describe('addRemote', () => {
  beforeAll(() => {
    registerSnapshots(snapshots)
  })
  it('addRemote', async () => {
    // Setup
    const { dir, gitdir } = await makeFixture('test-addRemote')
    const remote = 'baz'
    const url = 'git@github.com:baz/baz.git'
    // Test
    await addRemote({ dir, gitdir, remote, url })
    const a = await listRemotes({ dir, gitdir })
    expect(a).toEqual([
      { remote: 'foo', url: 'git@github.com:foo/foo.git' },
      { remote: 'bar', url: 'git@github.com:bar/bar.git' },
      { remote: 'baz', url: 'git@github.com:baz/baz.git' }
    ])
  })
  it('missing argument', async () => {
    // Setup
    const { dir, gitdir } = await makeFixture('test-addRemote')
    const remote = 'baz'
    const url = undefined
    // Test
    let error = null
    try {
      await addRemote({ dir, gitdir, remote, url })
    } catch (err) {
      error = err
    }
    expect(error).not.toBeNull()
    expect(error.toJSON()).toMatchSnapshot()
  })
  it('invalid remote name', async () => {
    // Setup
    const { dir, gitdir } = await makeFixture('test-addRemote')
    const remote = '@{HEAD~1}'
    const url = 'git@github.com:baz/baz.git'
    // Test
    let error = null
    try {
      await addRemote({ dir, gitdir, remote, url })
    } catch (err) {
      error = err
    }
    expect(error).not.toBeNull()
    expect(error.toJSON()).toMatchSnapshot()
  })
})
