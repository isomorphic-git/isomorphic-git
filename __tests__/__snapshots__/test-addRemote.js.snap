// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addRemote invalid remote name 1`] = `
Object {
  "caller": "git.addRemote",
  "code": "InvalidRefNameError",
  "data": Object {
    "noun": "remote",
    "ref": "@{HEAD~1}",
    "suggestion": "-HEAD-1}",
    "verb": "add",
  },
  "message": "Failed to add remote \\"@{HEAD~1}\\" because that name would not be a valid git reference. A valid alternative would be \\"-HEAD-1}\\".",
}
`;

exports[`addRemote missing argument 1`] = `
Object {
  "caller": "git.addRemote",
  "code": "MissingRequiredParameterError",
  "data": Object {
    "function": "addRemote",
    "parameter": "url",
  },
  "message": "The function \\"addRemote\\" requires a \\"url\\" parameter but none was provided.",
}
`;
